metadata-ingestion/tests/unit/hex/test_api.py

```diff
@@ -0,0 +1,235 @@
+            "description": "A test component",
+            "type": "COMPONENT",
+            "createdAt": "2022-02-01T12:00:00.000Z",
+            "lastEditedAt": "2022-02-02T12:00:00.000Z",
+            "status": {"name": "Draft"},
+            "categories": [{"name": "Category2"}],
+            "sharing": {"collections": [{"collection": {"name": "Collection2"}}]},
+            "creator": {"email": "creator@example.com"},
+            "owner": {"email": "owner@example.com"},
+            "analytics": {
+                "appViews": {
+                    "allTime": 50,
+                    "lastSevenDays": 5,
+                    "lastFourteenDays": 10,
+                    "lastThirtyDays": 15,
+                },
+                "lastViewedAt": "2022-02-03T12:00:00.000Z",
+            },
+        }
+
+        hex_api = HexAPI(
+            token=self.token,
+            report=self.report,
+            base_url=self.base_url,
+        )
+
+        # Test mapping a component
+        result = hex_api._map_data(component_data)
+
+        # Verify the result
+        assert isinstance(result, Component)
+        assert result.id == "component1"
+        assert result.title == "Test Component"
+        assert result.description == "A test component"
+        assert result.created_at == datetime(2022, 2, 1, 12, 0, 0, tzinfo=timezone.utc)
+        assert result.last_edited_at == datetime(
+            2022, 2, 2, 12, 0, 0, tzinfo=timezone.utc
+        )
+        assert result.status and result.status.name == "Draft"
+        assert (
+            result.categories
+            and len(result.categories) == 1
+            and result.categories[0].name == "Category2"
+        )
+        assert (
+            result.collections
+            and len(result.collections) == 1
+            and result.collections[0].name == "Collection2"
+        )
+        assert result.creator and result.creator.email == "creator@example.com"
+        assert result.owner and result.owner.email == "owner@example.com"
+        assert (
+            result.analytics
+            and result.analytics.appviews_all_time == 50
+            and result.analytics.last_viewed_at
+            == datetime(2022, 2, 3, 12, 0, 0, tzinfo=timezone.utc)
+        )
+
+    @patch("datahub.ingestion.source.hex.api.requests.get")
+    def test_fetch_projects_http_error(self, mock_get):
+        mock_response = MagicMock()
+        mock_response.raise_for_status.side_effect = requests.exceptions.HTTPError(
+            "500 Server Error: Internal Server Error"
+        )
+        mock_get.return_value = mock_response
+
+        hex_api = HexAPI(
+            token=self.token,
+            report=self.report,
+            base_url=self.base_url,
+        )
+
+        with self.assertRaises(requests.exceptions.HTTPError):
+            list(hex_api.fetch_projects())
+
+        assert self.report.fetch_projects_page_calls == 1
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
this shouldn't be necessary - you can do `pytest <filename>`

Reply:
addressed in commit 65e360672267823769bc2e95a2006e34e4f70399